engine/apps/alerts/tasks/notify_user.py

```diff
@@ -232,17 +240,18 @@ def perform_notification(log_record_pk):
     from apps.base.models import UserNotificationPolicy, UserNotificationPolicyLogRecord
     from apps.telegram.models import TelegramToUserConnector
 
-    # TODO: remove this log line once done investigation
     task_logger.info(f"perform_notification: log_record {log_record_pk}")
 
-    log_record = UserNotificationPolicyLogRecord.objects.get(pk=log_record_pk)
+    try:
+        log_record = UserNotificationPolicyLogRecord.objects.get(pk=log_record_pk)
+    except UserNotificationPolicyLogRecord.DoesNotExist:
+        task_logger.warning(
+            f"perform_notification: log_record {log_record_pk} doesn't exist. Skipping remainder of task. "
+            "The alert group associated with this log record may have been deleted."
+        )
+        return
```

Comment:
main change. There's more discussion behind this change [here](https://raintank-corp.slack.com/archives/C04JCU51NF8/p1705533264599479)